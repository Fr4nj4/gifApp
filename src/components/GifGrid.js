import React from 'react';
import PropTypes from 'prop-types';
// import { useEffect } from 'react';
import { GifGridItem } from './GifGridItem';
import { useFetchGifs } from '../hooks/useFetchGifs';
// import { getGifs } from '../helpers/getGifs';

const GifGrid = ({category}) => {


    const { loading, data } = useFetchGifs(category);


    return (
        <>
        <h3 className= ' animate__animated animate__fadeInDown'>{category}</h3>
        {loading && <p>Loaging</p>}
        <div className="card-grid">

                {
                    data.map( img => {
                       return (<GifGridItem 
                        key={img.id} 
                        {...img} />)
                    })
                }

        </div>
        </>
        
    )
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}

export default GifGrid
