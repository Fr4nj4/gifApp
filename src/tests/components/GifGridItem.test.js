import {shallow} from 'enzyme';
import { GifGridItem } from '../../components/GifGridItem';



describe('test GifGridItem', () => {


    const properties = {
        title: "naruto shippuden GIF",
        url: "https://media3.giphy.com/media/4gsjHZMPXdlGo/giphy.gif?cid=b3da8313ttx2xt6c4wdcqokp6mxlu2l93im34kr66cvylucq&rid=giphy.gif&ct=g"
    }

    const wrapper = shallow(<GifGridItem  {...properties}/>);

    test('should render item correctly ', () => {

     
        
        expect(wrapper).toMatchSnapshot();
    });
    

    test('should contain title', () => {
        const p = wrapper.find('p');
        expect(p.text().trim()).toBe(properties.title);

    });

    test('should url image equals to url properties', () => {
        const img = wrapper.find('img');
        
        expect(img.prop('src')).toBe(properties.url);
        expect(img.prop('alt')).toBe(properties.title);

    });


    test('should contains image class', () => {

        const div = wrapper.find('div');
        const className = 'card animate__animated animate__fadeInUp';
        expect(div.prop('className')).toBe(className);

        expect( div.prop('className').includes('animate__fadeInUp') ).toBe(true);

        
    })
    
    
    
    
})
