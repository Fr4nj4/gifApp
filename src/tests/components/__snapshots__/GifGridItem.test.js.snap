// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test GifGridItem should render item correctly  1`] = `
<div
  className="card animate__animated animate__fadeInUp"
>
  <img
    alt="naruto shippuden GIF"
    src="https://media3.giphy.com/media/4gsjHZMPXdlGo/giphy.gif?cid=b3da8313ttx2xt6c4wdcqokp6mxlu2l93im34kr66cvylucq&rid=giphy.gif&ct=g"
  />
  <p>
    naruto shippuden GIF
  </p>
</div>
`;
